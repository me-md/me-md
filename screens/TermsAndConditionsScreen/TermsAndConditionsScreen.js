import React, { useState, Fragment } from 'react';
import { Dimensions, Image, StyleSheet, Text, View } from 'react-native';
import { Button } from 'native-base';
import { CheckBox } from 'react-native-elements';
import { Header } from '../../components/Header';
import { ScrollView } from 'react-native-gesture-handler';
import { AntDesign } from '@expo/vector-icons';

const height = Dimensions.get('window').height;

export default function TermsAndConditionsScreen({ navigation }) {
  const [isChecked, setIsChecked] = useState({
    checked: false,
    isButtonEnabled: false
  });

  return (
    <Fragment>
      <Header />
      <ScrollView style={styles.container}>
        <View style={styles.contentContainer}>
          <AntDesign
            name='close'
            size={24}
            color='black'
            style={styles.close}
            onPress={() => navigation.push('Welcome')}
          />
          <Image
            source={require('../../assets/images/terms-and-conditions.png')}
            style={styles.icon}
          />
          <Text style={styles.heading}>Terms and Conditions</Text>
          <View style={styles.scrollContainer}>
            <ScrollView style={styles.scrollText}>
              <Text style={styles.legal}>
                This app is not intended to be a substitute for professional
                medical advice, diagnosis, or treatment. Always read the label
                before taking any over-the-counter (OTC) medications. The label
                identifies the active ingredient(s) and contains other important
                information including warnings about possible drug interactions
                and side effects. Always seek the advice of your physician or
                other qualified health provider with any questions you may have
                regarding a medical condition. Never disregard professional
                medical advice or delay in seeking it because of something you
                have read on meMD! If you think you may have a medical
                emergency, call your doctor or 911 immediately. meMD does not
                recommend or endorse any specific products or services. Reliance
                on any information provided by meMD is solely at your own risk.
              </Text>
            </ScrollView>
          </View>
          <View>
            <CheckBox
              center
              title={
                <Text style={styles.agreement}>
                  I read and accept the Terms of Service and Privacy Policy
                </Text>
              }
              checked={isChecked.checked}
              onPress={() =>
                setIsChecked({
                  checked: !isChecked.checked,
                  isButtonEnabled: !isChecked.isButtonEnabled
                })
              }
            />
          </View>
          {isChecked.isButtonEnabled ? (
            <Button
              block
              style={styles.button}
              onPress={() => navigation.push('BiologicalInformation')}
            >
              <Text>CONTINUE</Text>
            </Button>
          ) : (
              <Button disabled block style={styles.button}>
                <Text>CONTINUE</Text>
              </Button>
            )}
        </View>
      </ScrollView>
    </Fragment>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFFFFF',
    flex: 1
  },
  contentContainer: {
    alignItems: 'center',
    flex: 1
  },
  close: {
    alignSelf: 'flex-end',
    marginRight: height * 0.02,
    marginTop: height * 0.025
  },
  icon: {
    borderWidth: 0,
    borderRadius: 50,
    height: height * 0.1,
    width: height * 0.1
  },
  heading: {
    alignSelf: 'flex-start',
    margin: height * 0.02,
    marginBottom: height * 0.0005,
    fontSize: height * 0.04,
    fontWeight: 'bold'
  },
  scrollContainer: {
    height: height * 0.35,
    marginBottom: height * 0.0005
  },
  scrollText: {
    alignSelf: 'center',
    backgroundColor: '#CCCCCC',
    borderColor: '#FFFFFF',
    paddingLeft: height * 0.02,
    paddingRight: height * 0.02,
    margin: height * 0.02
  },
  legal: {
    lineHeight: height * 0.035,
    marginBottom: height * 0.02,
    marginTop: height * 0.02
  },
  agreement: {
    alignSelf: 'flex-start',
    fontSize: 14,
    margin: height * 0.015
  },
  button: {
    alignSelf: 'center',
    marginBottom: height * 0.03,
    marginTop: height * 0.02,
    width: '75%'
  }
});
