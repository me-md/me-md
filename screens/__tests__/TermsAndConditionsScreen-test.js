import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import TermsAndConditionsScreen from '../../screens/TermsAndConditionsScreen/TermsAndConditionsScreen';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
}));

describe('TermsAndConditionsScreen', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`should match a snapshot`, () => {
    const tree = renderer.create(<TermsAndConditionsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
