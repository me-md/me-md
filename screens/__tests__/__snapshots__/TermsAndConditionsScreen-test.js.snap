// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsAndConditionsScreen should match a snapshot 1`] = `
Array [
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278210303,
          4287155441,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 200.1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "borderWidth": 0,
          "height": 133.4,
          "marginBottom": 13.34,
          "marginTop": 60.03,
          "width": 133.4,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>,
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.8,
          }
        }
      >
        <Text />
        <Image
          source={1}
          style={
            Object {
              "borderRadius": 50,
              "borderWidth": 0,
              "height": 133.4,
              "width": 133.4,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "alignSelf": "center",
                "fontSize": 52.50000000000001,
                "fontWeight": "bold",
                "margin": 26.68,
                "marginBottom": 0.667,
              },
            ]
          }
          uppercase={false}
        >
          Terms and Conditions
        </Text>
        <View
          style={
            Object {
              "height": 466.9,
              "marginBottom": 0.667,
            }
          }
        >
          <RCTScrollView
            collapsable={false}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#CCCCCC",
                "borderColor": "#FFFFFF",
                "margin": 26.68,
                "paddingLeft": 26.68,
                "paddingRight": 26.68,
              }
            }
          >
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "fontSize": 24.011999999999997,
                      "lineHeight": 46.690000000000005,
                      "marginBottom": 26.68,
                      "marginTop": 26.68,
                    },
                  ]
                }
                uppercase={false}
              >
                This app is not intended to be a substitute for professional medical advice, diagnosis, or treatment. Always read the label before taking any over-the-counter (OTC) medications. The label identifies the active ingredient(s) and contains other important information including warnings about possible drug interactions and side effects. Always seek the advice of your physician or other qualified health provider with any questions you may have regarding a medical condition. Never disregard professional medical advice or delay in seeking it because of something you have read on MeMD! If you think you may have a medical emergency, call your doctor or 911 immediately. MeMD does not recommend or endorse any specific products or services. Reliance on any information provided by MeMD is solely at your own risk.
              </Text>
            </View>
          </RCTScrollView>
        </View>
        <View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#ccc",
                  "borderLeftWidth": 0.5,
                  "borderRadius": 2,
                  "borderRightWidth": 0.5,
                  "borderTopWidth": 0.5,
                  "elevation": 3,
                  "flexWrap": "nowrap",
                  "marginBottom": 5,
                  "marginLeft": 2,
                  "marginRight": 2,
                  "marginTop": 5,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 1.5,
                },
                Object {
                  "flex": 0.2,
                  "justifyContent": "space-evenly",
                  "width": 690,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "flex": 1,
                  },
                  Object {
                    "alignItems": "center",
                    "flex": 0.3,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                uppercase={false}
              >
                I have read and accept the Terms of Service
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#b5b5b5",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginBottom": 40.019999999999996,
              "marginTop": 26.68,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "width": "75%",
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            CONTINUE
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
